hls-plugin-api/src/Ide/Plugin/Config.hs

```diff
@@ -41,48 +48,63 @@ getInitialConfig (RequestMessage _ _ _ InitializeParams{_initializationOptions =
 -- | We (initially anyway) mirror the hie configuration, so that existing
 -- clients can simply switch executable and not have any nasty surprises.  There
 -- will be surprises relating to config options being ignored, initially though.
-data Config =
-  Config
-    { hlintOn                     :: Bool
-    , diagnosticsOnChange         :: Bool
-    , maxNumberOfProblems         :: Int
-    , diagnosticsDebounceDuration :: Int
-    , liquidOn                    :: Bool
-    , completionSnippetsOn        :: Bool
-    , formatOnImportOn            :: Bool
-    , formattingProvider          :: T.Text
-    } deriving (Show,Eq)
+data Config
+  = DefaultConfig
+  | UserConfig Bool Bool Int Int Bool Bool Bool T.Text
```

Comment:
This is not good enough. Please restore the field names or use Haddock comments to label the field